modules/influxdb/influxdb.go

```diff
@@ -0,0 +1,144 @@
+package influxdb
+
+import (
+	"context"
+	"fmt"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"path"
+	"strings"
+)
+
+// defaultImage {
+const defaultImage = "influxdb:1.8"
+
+// }
+
+// InfluxDbContainer represents the MySQL container type used in the module
+type InfluxDbContainer struct {
+	testcontainers.Container
+}
+
+// RunContainer creates an instance of the MySQL container type
+func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomizer) (*InfluxDbContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        defaultImage,
+		ExposedPorts: []string{"8086/tcp", "8088/tcp"},
+		Env: map[string]string{
+			"INFLUXDB_BIND_ADDRESS":          ":8088",
+			"INFLUXDB_HTTP_BIND_ADDRESS":     ":8086",
+			"INFLUXDB_REPORTING_DISABLED":    "true",
+			"INFLUXDB_MONITOR_STORE_ENABLED": "false",
+			"INFLUXDB_HTTP_HTTPS_ENABLED":    "false",
+			"INFLUXDB_HTTP_AUTH_ENABLED":     "false",
+		},
+	}
+	genericContainerReq := testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	}
+	for _, opt := range opts {
+		opt.Customize(&genericContainerReq)
+	}
+	hasInitDb := false
+	for _, f := range genericContainerReq.Files {
+		if f.ContainerFilePath == "/" && strings.HasSuffix(f.HostFilePath, "docker-entrypoint-initdb.d") {
+			// Init service in container will start influxdb, run scripts in docker-entrypoint-initdb.d and then
+			// terminate the influxdb server, followed by restart of influxdb.  This is tricky to wait for, and
+			// in this case, we are assuming that data was added by init script, so we then look for an
+			// "Open shard" which is the last thing that happens before the server is ready to accept connections.
+			// This is probably different for InfluxDB 2.x, but that is left as an exercise for the reader.
+			genericContainerReq.WaitingFor = wait.ForAll(
+				wait.ForLog("influxdb init process in progress..."),
+				wait.ForLog("Server shutdown completed"),
+				wait.ForLog("Opened shard"),
+				wait.ForExposedPort())
+			hasInitDb = true
+			break
+		}
+	}
+	if !hasInitDb {
+		if lastIndex := strings.LastIndex(genericContainerReq.Image, ":"); lastIndex != -1 {
+			tag := genericContainerReq.Image[lastIndex+1:]
+			if tag == "latest" || tag[0] == '2' {
+				genericContainerReq.WaitingFor = wait.ForLog(`Listening log_id=[0-9a-zA-Z_]+ service=tcp-listener transport=http`).AsRegexp()
+			}
+		} else {
+			genericContainerReq.WaitingFor = wait.ForLog("Listening for signals")
+		}
+	}
+	container, err := testcontainers.GenericContainer(ctx, genericContainerReq)
+	if err != nil {
+		return nil, err
+	}
+	return &InfluxDbContainer{container}, nil
+}
+
+func (c *InfluxDbContainer) MustHaveConnectionUrl(ctx context.Context, tls bool) string {
+	connectionString, err := c.ConnectionUrl(ctx, tls)
+	if err != nil {
+		panic(err)
```

Comment:
Do we want to panic here in the module code, or simply raise the error?

Reply:
Oh wait, this is a MustHave method, I guess to avoid the if err != nil, right? Forget about my previous comment.

I wonder if this pattern is common, as we do not usually use it, although I could see the value on it. If you think we can start adopting it, I'd appreciate more feedback around its usage 🙏

Reply:
The Must pattern is quite common in Go.  For example, regexp.MustCompile or the template.Must() wrapper.  It's useful to avoid the temporary err and if-then-panic.  If the caller doesn't want the panic, they simply call ConnectionUrl and handle the error themselves.

Reply:
Thanks for the explanation. I like this pattern, so we can start adopting it more and more 👍

Reply:
In any case, lets add the proper method comment explaining the usage and the expected behaviours, as in regex: https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/regexp/regexp.go;l=313-315